src/python/pants/backend/nfpm/fields/all.py

```diff
@@ -0,0 +1,177 @@
+
+from enum import Enum
+from typing import ClassVar
+
+from pants.core.goals.package import OutputPathField
+from pants.engine.target import Dependencies, StringField
+from pants.util.docutil import bin_name
+from pants.util.strutil import help_text
+
+
+class NfpmPackageNameField(StringField):
+    nfpm_alias = "name"
+    alias: ClassVar[str] = "package_name"
+    required = True
+    value: str
+    help = help_text(
+        """
+        The package name.
+        """
+    )
+
+
+class NfpmDependencies(Dependencies):
+    nfpm_alias = ""  # doesn't map directly to a nfpm.yaml field
+
+
+class NfpmArch(Enum):
+    # nFPM uses arch = GOARCH + GOARM. See: https://nfpm.goreleaser.com/goarch-to-pkg/
+    # GOARCH possible values come from `okgoarch` var at:
+    # https://github.com/golang/go/blob/go1.20.3/src/cmd/dist/build.go#L62-L79
+    # GOARM possible values come from `goarm()` func at:
+    # https://github.com/golang/go/blob/go1.20.3/src/internal/buildcfg/cfg.go#L77-L84
+    _386 = "386"
+    all = "all"  # not in `okgoarch`, but a conventionally accepted GOARCH value.
+    amd64 = "amd64"
+    arm5 = "arm5"  # GOARCH=arm GOARM=5
+    arm6 = "arm6"  # GOARCH=arm GOARM=6
+    arm7 = "arm7"  # GOARCH=arm GOARM=7
+    arm64 = "arm64"  # GOARCH=arm64
+    loong64 = "loong64"
+    mips = "mips"
+    mipsle = "mipsle"
+    mips64 = "mips64"
+    mips64le = "mips64le"
+    ppc64 = "ppc64"
+    ppc64le = "ppc64le"
+    riscv64 = "riscv64"
+    s390 = "s390"  # not in `okgoarch`; nFPM translates it to "s390x".
+    s390x = "s390x"
+    sparc64 = "sparc64"
+    wasm = "wasm"
+
+
+class GoOS(Enum):
+    # GOOS possible values come from `okgoos` var at:
+    # https://github.com/golang/go/blob/go1.20.3/src/cmd/dist/build.go#L81-L98
+    darwin = "darwin"
+    dragonfly = "dragonfly"
+    illumos = "illumos"
+    ios = "ios"
+    js = "js"
+    linux = "linux"  # apk, archlinux - no other value accepted
+    android = "android"
+    solaris = "solaris"
+    freebsd = "freebsd"
+    nacl = "nacl"
+    netbsd = "netbsd"
+    openbsd = "openbsd"
+    plan9 = "plan9"
+    windows = "windows"
+    aix = "aix"
+
+
+class NfpmArchField(StringField):
+    nfpm_alias = "arch"
+    alias: ClassVar[str] = nfpm_alias
+    # valid_choices = NfpmArch  # no valid_choices to support pass-through values.
+    default = NfpmArch.amd64.value  # based on nFPM default
+    help = help_text(
+        """
+        The package architecture.
+
+        This should be a valid GOARCH value (or GOARCH+GOARM) that nFPM can
+        translate into the package-specific equivalent. Otherwise, pants tells
+        nFPM to use this value as-is.
+        """
+    )
+
+
+class NfpmPlatformField(StringField):
+    nfpm_alias = "platform"
+    alias: ClassVar[str] = nfpm_alias
+    # valid_choices = GoOS  # no valid_choices to support pass-through values.
+    default = GoOS.linux.value  # based on nFPM default
+    help = help_text(
+        f"""
+        The package platform or OS.
```

Comment:
Add a link in this help text to https://nfpm.goreleaser.com/goarch-to-pkg/

(which is now correct: https://github.com/goreleaser/nfpm/pull/733).